flink-runtime/src/test/java/org/apache/flink/runtime/testutils/MiniClusterResource.java

```diff
@@ -253,12 +265,44 @@ private void startMiniCluster() throws Exception {
      * TaskManagerConf. Configurations which relates to streaming should be randomized in
      * TestStreamEnvironment#randomizeConfiguration.
      */
-    private static void randomizeConfiguration(Configuration configuration) {
+    private void randomizeConfiguration(Configuration configuration) throws IOException {
         // randomize ITTests for enabling buffer de-bloating
         if (RANDOMIZE_BUFFER_DEBLOAT_CONFIG
                 && !configuration.contains(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED)) {
             randomize(configuration, TaskManagerOptions.BUFFER_DEBLOAT_ENABLED, true, false);
         }
+
+        // randomize ITTests for enabling state change log
+        if (isConfigurationSupportedByChangelog(configuration)) {
+            if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_ON)) {
+                if (!configuration.contains(ENABLE_STATE_CHANGE_LOG)) {
+                    configuration.set(ENABLE_STATE_CHANGE_LOG, true);
+                }
+            } else if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_RAND)) {
+                if (randomize(configuration, ENABLE_STATE_CHANGE_LOG, true, false)) {
```

Comment:
The disadvantage of having the randomization here instead of [TestStreamEnvironment.java] is to randomize only once for one test class while TestStreamEnvironment randomizes for each test independently.

Reply:
IIUC, another issue is that if `MiniCluster` is used directly (i.e. without `MiniClusterResource`), the randomization won't be triggered. If that's the case, I think it's better to move it back to `TestStreamEnvironment` to have a better coverage.

Besides that, having all randomizations in a single place is more straightforward to analyze.